api/v1beta2/foundationdbcluster_types.go

```diff
@@ -576,15 +568,14 @@ func (processGroupStatus *ProcessGroupStatus) addCondition(oldProcessGroups []*P
 		}
 	}
 
-	// Check if we got a condition for the condition type for the ProcessGroupStatus
-	if oldProcessGroupStatus != nil {
-		for _, condition := range oldProcessGroupStatus.ProcessGroupConditions {
-			if condition.ProcessGroupConditionType == conditionType {
-				// Assumption: processGroupStatus doesn't have the conditionType
-				processGroupStatus.ProcessGroupConditions = append(processGroupStatus.ProcessGroupConditions, condition)
-				return
-			}
+	// If a process group is marked for removal and is fully excluded we only keep the ResourcesTerminating condition.
+	if processGroupStatus.IsMarkedForRemoval() && processGroupStatus.IsExcluded() {
+		if conditionType != ResourcesTerminating {
+			return
```

Comment:
Add a log statement here explaining why the condition is not being set?

Reply:
That sounds good.

Reply:
I haven't added a log statement here as this would probably noisy and we would only want to log this at the debug level and we would need to pass down the logger from all calls.